editor/sidebar/featured-image/index.js

```diff
@@ -52,21 +53,17 @@ class FeaturedImage extends Component {
 			return;
 		}
 		this.setState( { loading: true } );
-		const mediaIdToLoad = this.props.featuredImageId;
-		this.fetchMediaRequest = new wp.api.models.Media( { id: mediaIdToLoad } ).fetch()
+		if ( this.fetchMediaRequest ) {
+			this.fetchMediaRequest.abort();
+		}
+		this.fetchMediaRequest = new wp.api.models.Media( { id: this.props.featuredImageId } ).fetch()
 			.done( ( media ) => {
-				if ( this.props.featuredImageId !== mediaIdToLoad ) {
-					return;
-				}
 				this.setState( {
 					loading: false,
 					media,
 				} );
 			} )
 			.fail( () => {
-				if ( this.props.featuredImageId !== mediaIdToLoad ) {
```

Comment:
Cool 👍  So to be clear, neither the `done` nor `fail` case are invoked when the request is aborted?